src/cargo/util/config/target.rs

```diff
@@ -64,18 +64,62 @@ pub(super) fn load_target_cfgs(config: &Config) -> CargoResult<Vec<(String, Targ
     Ok(result)
 }
 
+/// Returns true if the `[target]` table should be applied to host targets.
+pub(super) fn get_target_applies_to_host(config: &Config) -> CargoResult<bool> {
+    if config.cli_unstable().target_applies_to_host {
+        if let Ok(target_applies_to_host) = config.get::<bool>("target-applies-to-host") {
+            Ok(target_applies_to_host)
+        } else {
+            Ok(!config.cli_unstable().host_config)
+        }
+    } else {
+        if config.cli_unstable().host_config {
+            anyhow::bail!(
+                "the -Zhost-config flag requires the -Ztarget-applies-to-host flag to be set"
+            );
+        } else {
+            Ok(true)
+        }
+    }
+}
+
+/// Loads a single `[host]` table for the given triple.
+pub(super) fn load_host_triple(config: &Config, triple: &str) -> CargoResult<TargetConfig> {
+    if config.cli_unstable().host_config {
```

Comment:
I'm not sure if this `if` is necessary since it seems like most of the gating happens in `get_target_applies_to_host`?

Reply:
So `get_target_applies_to_host` is for gating the availability of the `target-applies-to-host` config option, this is for gating the `host-table` config option which is separate.